### PR TITLE
Fix photo route

### LINKED ISSUE #301: Users need to understand that you can't use a paper photo
If you answer that you already have a paper photo, you never get the option to use a tablet or a smartphone to take a picture.

### PR DESCRIPTION
**Pull request by suryanaga**
Fixes #301.

## Before

image

## After

image

## And with mobile detection
image

I've also added this flow chart to the pattern text.

### CODE REVIEW

**Comment by suryanaga** [comment]
@petegale If I recall correctly, you had some more comments about this - Could you add them here please? Thanks.

**Comment by petegale** [comment]
Yes, I'm not sure if the new logic works for the common case where someone is using a laptop to complete the form and a smartphone to take their photo

**Comment by suryanaga** [comment]
Added the word 'digital' in two places to address #466. 

screenshot-localhost-3000 2017-04-28 13-26-31

**Comment by suryanaga** [comment]
Added a sentence about a digital camera including being on a smartphone to address the issue highlighted by @petegale's above comment. 

screen shot 2017-04-28 at 13 42 10

**Comment by lizziebruce** [comment]
@petegale the wording changes here are as per my recommendations. I've since thought that "mobile phone" may be more widely understood by users than "smartphone" – what do you think?? Regarding the journey flow I don't think the questions work well at all, as observed in UR. Am revisiting these pages in my content review – will prioritise and send suggestions.

**Comment by suryanaga** [comment]
Not all mobile phones have a camera, and I think "smartphone" is a widely understood term.

**Comment by lizziebruce** [comment]
@sanjaypoyzer Did you read the sentence again, replacing smartphone with mobile phone? It reads: "This could include having a camera on your mobile phone or tablet." 
My question about how widespread the term "smartphone" is is for @petegale. I'm not convinced it's understood by this demographic. None of the users I observed used the term smartphone, however several said mobile phone.

**Comment by lizziebruce** [comment]
@irinapencheva Content redesign to replace all of the pre-'Confirm your photo' photo pages which confused users. Perhaps we can use this as a starting point for a more streamlined journey. The page is now quite long, especially if Help sections are expanded. But if gives users all they need to know to take a photo and avoids a binary choices tree flow for a multiple choice task.. 

screen shot 2017-04-28 at 15 13 26

@irinapencheva reflagging as discussed

**Comment by suryanaga** [comment]
@lizziebruce 

Re: Smart phone vs mobile phone - You're right, I didn't read the sentence through with the replacement, sorry. I think you're also right on reflection that mobile phone is definitely used by more people so I'll change that.

---

Re: Getting rid of the questions - That's quite a big departure from the current design, and I would say outside the scope of this pull request. In general, it's best to be referencing issues when suggesting changes as they are our documents of user research. 

We added the questions because there's too many different choices and too much information to make a short page, and we know that when users are faced with a long page of information, they will get overwhelmed and not read it. As far as I know, right now we don't have any documented user research that suggests a move away from the questions is necessary and I think it performs better with users than a page trying to do all of the routing would do. 

In short, it's less for the user to think about and process mentally this way.

**Comment by petegale** [comment]
@sanjaypoyzer @lizziebruce coming to this a little late, but I've done quite a lot of work on smartphone vs mobile phone. Generally mobile phone is a more widely understood concept but includes simpler phones. In the research I did, we found smartphone was the best way to refer to the sort of phone that you can install apps on. I would agree that in this instance, it's probably best to use mobile phone. There will be a very small number of users with mobile phones without cameras, but they will probably be quite aware of this limitation

**Comment by petegale** [comment]
@sanjaypoyzer  (as we have just discussed )have taken a look at this and wanted to make a few points
1. We appear to refer to the user's computer or device inconsistently (have you got a camera on this device vs. is your photo on this computer) I'm unsure how people will respond to the word device, and why we're inconsistent here.
2. We still use the H1 'do you have a digital camera' but then qualify it by saying 'This could include having a camera on your mobile phone or tablet.'. We've seen that 'digital camera' is interpreted as meaning an actual camera. Perhaps we should change the H1 to say 'Can you take a digital photo' to address this inconsistency?
3. Third and final thought - If we assume that we can auto-detect whether the user's device has a camera or not, isn't it then possible to simplify this interaction. Have a camera - just take a photo / Don't have a camera -> have a smartphone or tablet? -> Let us send you a link to take a photo

**Comment by suryanaga** [comment]
I'm not sure about point 1 "Can you take a digital photo?" as we've seen some people are very unconfident with this, so I'm worried people will say no even if they do have access to a camera


For other two:

 - [x] I'll detect mobile devices using [mobile-detect.js](https://hgoebl.github.io/mobile-detect.js/) or similar 
 - [x] Display "device" for mobiles and tablets and "computer" for others (addressing point 2)
 - [x] Take users straight to the "Take a photo" page if they are on a device that is likely to have a camera (addressing point 3)
 - [x] Add a link on the "Take a photo" page to get to the other one if we incorrectly route the user there (through wrong question/answer routing, or through wrong device recognition) 

The point about sending a link to users phones for uploading feels a bit outside the scope of this PR, so will leave it out for a future one, leaving the upload process untouched.

**Comment by lizziebruce** [comment]
@sanjaypoyzer @petegale @irinapencheva re questions, there aren't any user issues/quotes mentioned at the top of this issue, so thought it was about the whole flow diagram.. I can now see #301 is referenced. 
Appreciate the long page isn't ideal – I purposefully showed how the opposite extreme would look to help us rethink this segment of the journey. Thinking it needs to be questions seems to have got cemented and the user testing didn't show great results for this. So suggested this alternative as a starting point for revisiting this segment of the journey. The questions in the flow chart are quite prescriptive and don't cover all scenarios (e.g. do you have a relative or friend with a camera on their device). Also "device" isn't necessarily an obvious word for many of the population.

**Review by ghost — COMMENTED** [review]
Leaving feedback without approving

**Review by ghost — DISMISSED** [review]
Approving changes made

**Comment by suryanaga** [comment]
This is the new flow with @petegale's suggestion:

 
screen shot 2017-05-04 at 12 41 15

Note: There's a link on the "Take a photo" page which allows them to say they don't have a camera on this device, mainly because the mobile detection isn't perfect. That link takes them to "Do you have a digital camera?"

screenshot-localhost-3000 2017-05-04 12-47-28

I've also made it display the word "device" instead of "computer" on devices that are detected as mobile:
screen shot 2017-05-04 at 12 48 39

I've done the same device/computer switcheroo on the page which asks "Do you have a camera on your device?"
screen shot 2017-05-04 at 12 52 34

There aren't any other content changes in this pull request.

Now awaiting 👍 from @lizziebruce and @irinapencheva.

**Comment by irinapencheva** [comment]
@sanjaypoyzer - I'm happy with the overall flow. 

A few points / questions:

1. Shouldn't there be a consent / question somewhere to confirm the user is happy to take the photo with the webcam (i.e. before the camera devise is launched for privacy purposes). This might be covered by the screen 'Take a photo with this device' and maybe I just can't see it.

2. Going back to the consistent use of 'device' vs. computer on the decision point after 'Do you have a digital photo?' --> NO shouldn't the follow up ? be 'Do you have a camera on this computer?' rather than 'this device'? Wouldn't the majority of the people who reach that point will still be using a computer if they've not been detected originally?

3. As per comments above - shouldn't we rephrase 'digital camera' as this is usually not understood to be a mobile, which many people will tend to use?

**Comment by suryanaga** [comment]
@irinapencheva 

1. Not sure what you mean by "consent / question". 
2. It says device if it's a mobile device, computer if not. 
3. The page which mentions digital camera also says that this counts a camera on a phone, as shown in above screenshot. Here it is again:

screen shot 2017-05-04 at 15 11 13

**Review by irinapencheva — APPROVED** [review]

**Comment by petegale** [comment]
@lizziebruce I wanted to flag some things I've observed during research, as I think they might inform the content on the "Do you have a digital camera" page.

Two things I've observed in research:
First is that I've seen people interpret the question 'Do you have a digital camera' as meaning an actual camera, and not a device that has a camera, such as a tablet or a smartphone. This means that these users will have to read the qualifying text to be able to answer this question correctly.

Secondly, I've seen a lot of people scan reading simple pages such as this, and missing the sort of qualifying text we have under the heading.

I was wondering if it might be better to ask about whether or not they're able to take a digital photo? As that avoids the problem of interpretation

**Comment by lizziebruce** [comment]
@petegale @sanjaypoyzer @irinapencheva  the best solution would be to include mobile etc in the radio button text. I'll come up with some new copy for this.

**Comment by lizziebruce** [comment]
@petegale @sanjaypoyzer @irinapencheva 

> 
> # Do you have a digital camera?
> 
> - Yes, I have a digital camera or a mobile phone, tablet or laptop which has a camera
> - No, I don't have a digital camera

OR

> # Do you have a digital camera?
> 
> - Yes, I have a digital camera 
> - No, but I have a mobile phone, tablet or laptop which has a camera
> - No, I don't have a digital camera or a device with a digital camera

In the 2nd version, selecting the middle radio button would take user to same page as selecting the top radio button.

Should we also ask them if they have someone around who can take the photo, since we're stipulating that someone else takes it for necessary distance? E.g.

> # Is there someone who can take a photo of you now with your digital camera or device?
> 
> - Yes 
> - No

**Comment by suryanaga** [comment]
@lizziebruce I think the second option is better, but I'm worried that the "No" in the second option would discourage users, making them think this option wouldn't be acceptable, and maybe even make them abandon the service without choosing any option.

What do you think about removing the "No," ? And the "Yes" from the first option?

So it would be:

> ## Do you have a digital camera?
> 
> -  I have a digital camera
> - I have a mobile phone, tablet or laptop which has a camera
> - No, I don't have a digital camera or a device with a digital camera

**Comment by irinapencheva** [comment]
@lizziebruce - agree with @sanjaypoyzer on the Y/N and optioning

**Comment by lizziebruce** [comment]
@irinapencheva @sanjaypoyzer I'm happy for the yes/nos to be taken out. They were a legacy from the previous copy. 

What do you both think of the additional question re: Is there someone who can take a photo of you now with your digital camera or device?

As we are asking people get someone else to take the photo rather than take a selfie, therefore if no-one else is around even if they have a digi camera / device with one ultimately they won't be able to produce the photo.

**Review by lizziebruce — CHANGES_REQUESTED** [review]
Do you have a digital camera?

I have a digital camera
I have a mobile phone, tablet or laptop which has a camera
No, I don't have a digital camera or a device with a digital camera

**Comment by irinapencheva** [comment]
@lizziebruce - I think selfies are also acceptable if they can fit the general criteria, so I'd be inclined not to add this question. Or maybe not have it as a question but some kind of guidance/ sentence below to say "You can also ask someone to take ..."

**Comment by lizziebruce** [comment]
@irinapencheva @sanjaypoyzer fine, was confusing with the passport photo criteria showcased in the Design training last week, sorry. (However given the demographic users may experience problems taking a selfie!)
Suggest add:

> "You can ask someone else to take your photo or try taking one of yourself." 

before "Your photo doesn't need to be professionally taken" on page below and include guidance on both options in the Get help link destination content.

screen shot 2017-05-10 at 13 00 47

**Comment by suryanaga** [comment]
@lizziebruce Done: 

image

image

**Comment by suryanaga** [comment]
Now just waiting for @samuelculley to check for bugs and merge.

Note for Sam: You'll need to run `npm install` before you run `npm start` because I've added a node module to do the mobile detecting.

**Review by lizziebruce — APPROVED** [review]

**Review by ghost — APPROVED** [review]
All seems to be working, was able to move through without hitting any weird loops. Only thing I'm unsure about is there being some links that don't go anywhere, but that can be addressed later!

Screenshot below just shows a few of the links that don't work - clicking on "Get help taking a photo" just returns you to the top of the page. 

image
